src/components/SearchBar.js

```diff
@@ -1,15 +1,51 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { searchPhoto } from '../actions';
 
-const SearchBar = () => {
-    return (
-        <div className="ui loading search">
-            <div className="ui icon input">
-                <input className="prompt" type="text" placeholder="Search..." />
-                <i className="search icon"></i>
-            </div>
-            <div className="results"></div>
-        </div>
-    );
+class SearchBar extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = { searchTerm: null };
+    }
+
+    onSearchChange = (event) => {
+        var searchTerm = event.target.value;
+
+        this.setState({ searchTerm });
+    }
+
+    onSearchSubmit = (event) => {
+        event.preventDefault();
+
+        this.props.searchPhoto(this.state.searchTerm);
+        this.props.onSearchSubmit(this.state.searchTerm);
+    }
+
+    render() {
+        return (
+            <form 
```

Comment:
If you'd really like to spice things up maybe try using redux-forms? I'm ok with the current implementation for now.

Reply:
I've read a few posts regarding redux-forms and it seemed pretty useful. I'll refactor the JSX around forms with redux-forms once we have the basic functionalities in place.